0015-ilp-addresses/0015-ilp-addresses.md

```diff
@@ -0,0 +1,123 @@
+# ILP Addresses
+
+_ILP addresses_ provide a way route payments to their intended destination through a recursive series of hops, including any number of ILP Connectors. (This happens after the payment is set up on a higher level, such as [SPSP](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md).) Addresses can be subdivided into two categories:
+
+- **Destination Addresses** are complete addresses that can receive payments. A destination address always maps to one account in a ledger. (It can also provide more specific information, such as an invoice ID or a sub-account.) Destination addresses MUST NOT end in a period (`.`) character.
+- **Prefix Addresses** are incomplete addresses representing a group of destination addresses. Many depths of grouping are possible: groups of accounts or sub-accounts, an individual ledger or subledger, or entire neighborhoods of ledgers. Prefix addresses MUST end in a period (`.`) character. Payment setup protocols MUST reject payments to prefix addresses.
+
+Both types of address usually contain one or more period characters as separators.
+
+## Address Requirements
+
+ILP Addresses must meet the following requirements:
+
+1. The address MUST begin with a prefix indicating the allocation scheme. The only supported allocation scheme for now is the [General Allocation Scheme](#general-allocation-scheme), which uses the prefix `g.`.
```

Comment:
I think we need to clarify what it means to "leave this open" -

- If you're saying that implementation should validate addresses that don't start with `g.`, I can see some rationale for that. However, any such implementation is not going to be able to know whether the addresses are following all the rules of the other allocation schemes.
- If you're saying that people should be able to claim addresses that don't start with `g.` but are otherwise the same as the generally-allocated addresses, I think that's unwise. That defeats the purpose of using a general allocation prefix.

When we devise other allocation schemes, it's easy to update this spec with more prefixes _and_ the rules of the other allocation schemes. That protects the future allocation schemes from having existing addresses occupying their namespace, and it protects implementations from trying to understand addresses that are allocated through a 

Imagine a "pubkey" allocation scheme, where the address is a public key that can be looked upon a keyserver - one of the rules of the allocation scheme is that the pubkey exists and isn't revoked on a consensus of key servers. We may or may not want implementations to route things to addresses in the `pubkey.` space if those implementations don't know the difference.

So I'm for leaving this as-is until we have additional allocation schemes, and officially defining other allocation schemes as they come into use.